packages/types/src/util/index.ts

```diff
@@ -5,6 +5,7 @@ export const LiteralSchema = z.union([
   z.number(),
   z.boolean(),
   z.null(),
+  z.undefined(),
```

Comment:
@AndrewCLu this was added because the JsonSchema was panicking, currently we convert null to undefined in UserData but JsonSchema currently doesn't allow undefined. Do you know if this will have any weird consequences?

Reply:
JSON doesn't allow undefined so I found a better solution -- ie in this case convert undefined back to null.